### PR TITLE
Added `netutilsparser` to `network_driver`

### PR DESCRIPTION
**Pull request by itdependsnetworks**
# Closes: DNE
# What's Changed

Add netutilsparser

image

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)

### CODE REVIEW

**Comment by glennmatthews** [comment]
Please target this PR to the `develop` branch rather than `ltm/1.6`. We should always be making code changes in `develop` first then making the decision as to whether it's justifiable to back-port them to the LTM branch after they've been accepted and merged.

File: nautobot/docs/release-notes/version-1.6.md
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-1.6.md:46]: This PR is for 2.0 not for 1.6, and this specific change is already covered by `changes/4396.added`.

```suggestion
The [Platform](../user-guide/core-data-model/dcim/platform.md) model has been enhanced to include a `network_driver` database field and a `network_driver_mappings` derived property based on the [`netutils`](https://netutils.readthedocs.io/en/latest/) library. For example, if you set a Platform to have a `network_driver` value of `"cisco_ios"`, the `platform.network_driver_mappings` property will return a dictionary containing `ansible`, `hier_config`, `napalm`, `netmiko`, `ntc_templates`, `pyats`, `pyntc`, and `scrapli` keys corresponding to this entry. These properties can be referenced via the REST API and GraphQL to assist in developing and maintaining Apps, Jobs, or third-party code that interact with devices by using any of these libraries.
```

File: pyproject.toml
Comment (glennmatthews) [inline pyproject.toml:112]: Let's add a `changes/4418.dependencies` documenting this change.

File: changes/4396.added
Comment (glennmatthews) [inline changes/4396.added:1]: Should be `changes/4418.added`.

File: nautobot/dcim/utils.py
Comment (glennmatthews) [inline nautobot/dcim/utils.py:102]: Let's document this in `nautobot/docs/user-guide/core-data-model/dcim/platform.md` as a 2.0 addition to the `Platform` model functionality.

File: nautobot/docs/release-notes/version-1.6.md
Comment (itdependsnetworks) [reply nautobot/docs/release-notes/version-1.6.md:46]: > This PR is for 2.0 not for 1.6, and this specific change is already covered by changes/4396.added.

I am not sure I follow, this was for 1.6, you asked me to set to develop and not to ltm branch, are you saying "we will not accept this for 1.6"? If so, fair enough, would be good to understand why though so in the future I can better understand what does/does not get into ltm.

Comment (glennmatthews) [reply nautobot/docs/release-notes/version-1.6.md:46]: `develop` is for 2.0 at this time, so this PR is currently making code changes to the upcoming 2.0 release. If there's a use case for including this feature in a 1.6.x patch release as well, we can deal with that after this PR is merged into `develop`, either as a cherry-pick effort by the Nautobot core team or by having someone open a second PR.

https://docs.nautobot.com/projects/core/en/stable/development/#maintenance-release-ltm is the current documentation regarding the plan for maintenance of the LTM train - short version is that LTM will be guaranteed security and data-loss bug fixes, everything else will be case-by-case. This PR is technically speaking a new feature rather than a bug fix, so personally I think that makes it less likely to be accepted for backporting to a 1.6.x patch release per our usual SemVer expectations, but all things are possible if there's a strong enough use case or need.

**Review by glennmatthews — APPROVED** [review]

**Comment by itdependsnetworks** [comment]
Safe to assume you will have the poetry conflict? This is a moving target if someone from core doesn't handle and then merge instantly.

**Pull request merged**
